openghg/standardise/_standardise.py

```diff
@@ -46,8 +46,8 @@ def standardise_surface(
     site_filepath: optionalPathType = None,
     store: Optional[str] = None,
     update_mismatch: str = "never",
-    if_exists: str = "default",
-    save_current: Optional[bool] = None,
+    if_exists: str = "auto",
```

Comment:
Just to check - what would the user need to pass for this? If it was just "auto" then that would be fine but the reason I didn't do this before was because I didn't want to the user to have to pass something more complex like Literal["auto"] - what do you think though?

Reply:
Oh no, a `Literal` type just means there's a limited set of strings the user should pass in, then the user might get those options suggested to them by their IDE.

Reply:
`Literal` is used for the situation where the user will (and must) pass in a *literal* string in that place. E.g. to force calling it like
```python
standardise_surface(..., if_exists="new")
```
It was introduced to allow SQL libraries to require that the function was only called with plain literal strings to stop people doing string interpolation themselves (and introducing potential injection vulnerabilities).

In this case, if you set the type to be a `Literal` it would prevent something like the following:

```python
method = "new"
standardise_surface(..., if_exists=method)
```

with an error like:
```
error: Argument to "standardise_surface" has incompatible type "str"; expected "Literal['auto', 'new', 'combine']"  [arg-type]
```

If you have a defined set of values, and you want to provide that information to the type system, the solution is an enum (though that might not be the interface that you want here, due to it being unfamiliar to your users and requires more boilerplate).

Reply:
Ah yes I think I was confusing `Enum` and `Literal` in my original comment.

Ok, so sounds like using a `Literal` may require may require a bit more thought so I think for now we should stick with the str and we can consider whether that may be a better approach if/when we work out some of the typing and mypy implications of using them.